crates/burn-core/src/nn/rnn/lstm.rs

```diff
@@ -39,164 +56,253 @@ impl LstmConfig {
             hidden_state = output_values * cell_state.clone().tanh();
 
-            let unsqueezed_shape = [cell_state.shape().dims[0], 1, cell_state.shape().dims[1]];
-
-            let unsqueezed_cell_state = cell_state.clone().reshape(unsqueezed_shape);
-            let unsqueezed_hidden_state = hidden_state.clone().reshape(unsqueezed_shape);
+            let unsqueezed_hidden_state = hidden_state.clone().unsqueeze_dim(1);
 
-            // store the state for this timestep
-            batched_cell_state = batched_cell_state.slice_assign(
-                [0..batch_size, t..(t + 1), 0..self.d_hidden],
-                unsqueezed_cell_state.clone(),
-            );
+            // store the hidden state for this timestep
             batched_hidden_state = batched_hidden_state.slice_assign(
                 [0..batch_size, t..(t + 1), 0..self.d_hidden],
                 unsqueezed_hidden_state.clone(),
             );
         }
 
-        (batched_cell_state, batched_hidden_state)
+        (
+            batched_hidden_state,
+            LstmState::new(cell_state, hidden_state),
+        )
+    }
+}
+
+/// The configuration for a [Bidirectional LSTM](BiLstm) module.
+#[derive(Config)]
+pub struct BiLstmConfig {
+    /// The size of the input features.
+    pub d_input: usize,
+    /// The size of the hidden state.
+    pub d_hidden: usize,
+    /// If a bias should be applied during the BiLstm transformation.
+    pub bias: bool,
+    /// BiLstm initializer
+    #[config(default = "Initializer::XavierNormal{gain:1.0}")]
+    pub initializer: Initializer,
+}
+
+/// The BiLstm module. This implementation is for Bidirectional LSTM.
+#[derive(Module, Debug)]
+pub struct BiLstm<B: Backend> {
+    /// LSTM for the forward direction.
+    pub forward: Lstm<B>,
+    /// LSTM for the reverse direction.
+    pub reverse: Lstm<B>,
+    d_hidden: usize,
+}
+
+impl BiLstmConfig {
+    /// Initialize a new [Bidirectional LSTM](BiLstm) module.
+    pub fn init<B: Backend>(&self, device: &B::Device) -> BiLstm<B> {
+        BiLstm {
+            forward: LstmConfig::new(self.d_input, self.d_hidden, self.bias)
+                .with_initializer(self.initializer.clone())
+                .init(device),
+            reverse: LstmConfig::new(self.d_input, self.d_hidden, self.bias)
+                .with_initializer(self.initializer.clone())
+                .init(device),
+            d_hidden: self.d_hidden,
+        }
     }
+}
 
-    /// Helper function for performing weighted matrix product for a gate and adds
-    /// bias, if any.
+impl<B: Backend> BiLstm<B> {
+    /// Applies the forward pass on the input tensor. This Bidirectional LSTM implementation
+    /// returns hidden state for each element in a sequence (i.e., across `seq_length`) and a final state,
+    /// producing 3-dimensional tensors where the dimensions represent `[batch_size, sequence_length, hidden_size * 2]`.
```

Comment:
Same as with the `Lstm` forward docstring. We can simplify to:

Applies the forward pass on the input tensor. This Bidirectional LSTM implementation returns the state for each element in a sequence (i.e., across `seq_length`) and a final state.